modules/common/types/Pattern.ts

```diff
@@ -46,5 +32,6 @@ export class Pattern<T extends PatternParameters> {
       },
       ...parameters,
     };
+    this.paramValues = ParamsProxy(this.params);
```

Comment:
This `ParamsProxy` does some funny stuff. It uses a `Proxy` object to present a different interface for getting+setting the parameter values, presenting the params object in what looks like just a single flat object. Getting and setting values on the proxy object get and set the equivalent nested values from the original object, allowing us to continue using the original object for uniforms.

For example, in the table below, the left params object would be accessible as a proxy object like the paramValues objects on the right, and getting and setting things on the paramValues would actually update the original more deeply nested params object.

<table>
<thead><tr><th>this.params (original)</th><th>this.paramValues (proxy of original)</th></tr></thead>
<tbody>
<tr><td>

```
{
    u_time: {
      name: "Time",
      value: 0,
    },
    u_global_time: {
      name: "Global Time",
      value: 0,
    },
    u_resolution: {
      name: "Resolution",
      value: Vector2(96, 75),
    },
    u_blah: {
      name: "Blah",
      value: 0,
    },
    u_a: {
      name: "A",
      value: Vector2(0,0),
    }
}
```
</td><td>

```
{
  u_time: 0,
  u_global_time: 0,
  u_resolution: Vector2(96, 75),
  u_blah: 0,
  u_a: Vector2(0,0)
},
```
</td></tr>
<tbody>
</table>
</table>

Reply:
Is this good? Dunno. I just think it's neat.

Reply:
This is pretty cool. Haven't run into these proxies before, but it seems to solve the issue we had so I'm down 👍

Reply:
Proxies be cool